src/descriptor/mod.rs

```diff
@@ -668,7 +670,12 @@ mod tests {
                 "512102000000000000000000000000000000000000000000000000000000000000000251ae"
             )
         );
-        assert_eq!(bare.address(bitcoin::Network::Bitcoin,), None);
+        assert_eq!(
+            bare.address(bitcoin::Network::Bitcoin)
+                .unwrap_err()
+                .to_string(),
+            "Bare descriptors don't have address"
```

Comment:
maybe comparing to `Error::BareDescriptorAddr.to_string()` is slightly better than repeating the string